src/connection-manager/connections.ts

```diff
@@ -55,11 +62,31 @@ async function handleWsMessageTypes(
     if (ledger_hashes.includes(validationData.ledger_hash)) {
       validationData.networks = networks
     }
-    void agreement.handleValidation(data as ValidationRaw)
+    const validationNetworkDb: DatabaseValidator | undefined = await query(
+      'validators',
+    )
+      .select('*')
+      .where('signing_key', validationData.validation_public_key)
+      .first()
+    const validationNetwork =
+      validationData.networks ?? validationNetworkDb?.networks
+    if (validationNetwork) {
+      validationData.ledger_fee = network_fee.get(validationNetwork)
+    }
+    void agreement.handleValidation(validationData)
```

Comment:
Why is the type assertion not needed here anymore?

Reply:
validationData already has type assertion on line 61: `const validationData = data as ValidationRaw`